### PR TITLE
VACMS-5492 - disable event cost field and update default value to free

### LINKED ISSUE #5492: Disable or hide the cost field on events
## User Story or Problem Statement

<details><summary>Background</summary>

Jane:
> Jean has expressed some concerns/risks that would be better addressed in the content model/form design than by training, such as not allowing people to include a cost for events (because VEO doesn’t ever want events with a cost to go into the calendar).
> She and Doug seemed keen to talk about it further, I told them I’d convey that back to the team.

Kev's response
> This has come up before. i think it’s a problem where the content strategy didn’t make it into the CMS UI. Research with Veterans showed that they still wanted to be told that an event was free.  There’s a form design question about whether or not to provide a default value “Free” and then disable the field, or just hide the field altogether. I can imagine a situation in the future  where a fundraiser event will happen and we’ll want to be able to make this field editable. In any case, i’ll document this as a problem that needs solving and add to the Event form UX epic.

</details>

<details><summary>Screenshot</summary>

Edit_Event_Bunker_Bingo___VA_gov_CMS

</details>

## Acceptance Criteria
- [ ] Make "Free" the default value
- [ ] Disable the field for non-content Admins
- [ ] Remove textcounter
- [ ] Help text should read "All VA events are free"

## Implementation steps
Drupal config and form theming.

## Design principles

Which [CMS design principle](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/cms/cms-product-design-principles.md) is at play?
- [ ] Purpose-driven: Creates an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] Efficient: Removes distractions and create clear, straightforward paths to get the job done.
- [ ] Approachable: Offers friendly guidance over authoritative instruction.
- [x] Consistent: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] Empowering: Provide clear information to help editors make decisions about their work.

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Core Application Team`
- [x] `Product Support Team`

### PR DESCRIPTION
**Pull request by msbtterswrth**
## Description

Closes #5492 .

## Testing done
Created an event and confirmed the 'cost' field was hidden.

## Screenshots
current: 
image

updated:
image

## QA steps

As cms editor I can:
- [x] 1. Create a new event (/node/add/event) and verify that the cost field does not appear in the form under the registration section.
- [x] 1. View the field settings for cost (/admin/structure/types/manage/event/fields/node.event.field_event_cost) and confirm the help text was updated and the default value set to 'free'
2. Then validate Acceptance Criteria from issue
- [x] Make "Free" the default value
- [x] Disable the field for non-content Admins (updating to disabled for ALL admins)
- [ ] Remove textcounter (i left this in since we're just disabling it completely)
- [x] Help text should read "All VA events are free" 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS team`
- [x] `Sitewide CMS Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [x] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by kevwalsh — CHANGES_REQUESTED** [review]
One of the ACs is 

> Help text should read "All VA events are free"

But i think this could be improved to say 

"All events on VA.gov are free."

Can you add that to the fieldgroup help text? 

<details><summary>Screenshot</summary>


Cursor_and_144301074-55c448d9-e5ef-4bfe-b307-7b1cae293d73_png__2745×1708_

</details>

**Comment by msbtterswrth** [comment]
@kevwalsh updated: 
image

File: config/sync/core.entity_form_display.node.event.default.yml [hunk 1]
```diff
@@ -86,7 +85,6 @@ third_party_settings:
     group_editorial_workflow:
       children:
         - moderation_state
-        - revision_log
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.event.default.yml:89]: Klinger is trying to go AWOL again. ;)
 Please add this back.   
Note: git add -i    is your friend when encouting changes to form_display.node.*.default.yml files.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
One fix needed.

File: config/sync/core.entity_form_display.node.event.default.yml (hunk 1 above)
Comment (msbtterswrth) [reply config/sync/core.entity_form_display.node.event.default.yml]: i swore i fixed this in my last commit, but alas it didn't go up. do we know why this ALWAYS disappears whenever you touch a node?

Comment (swirtSJW) [reply config/sync/core.entity_form_display.node.event.default.yml]: See thread  https://dsva.slack.com/archives/CT4GZBM8F/p1636469291454500

**Review by swirtSJW — APPROVED** [review]
Looks good.

One odd editorial issue: showing event is free in node view, but not having a place to set it.
image

File: config/sync/core.entity_form_display.node.event.default.yml [hunk 2]
```diff
@@ -70,13 +69,13 @@ third_party_settings:
...
       children:
         - field_event_registrationrequired
-        - field_event_cost
         - group_registration_link
         - field_additional_information_abo
       parent_name: ''
       weight: 6
       format_type: details
       format_settings:
+        description: 'All events on VA.gov are free.'
```
Comment (kevwalsh) [inline config/sync/core.entity_form_display.node.event.default.yml:78]: @swirtSJW here's the diff that addressed my request.

**Review by kevwalsh — APPROVED** [review]

**Comment by kevwalsh** [comment]
> Looks good.
> 
> One odd editorial issue: showing event is free in node view, but not having a place to set it. image

Maybe not ideal, but at least it mirrors the FE experience

**Comment by swirtSJW** [comment]
image

**Pull request merged**